Web Inspector: Console: internal properties are not greyed out in object previews

dcrousso opened the pull request:
#### 6d890dea244d3c6ab908b53ee3cfb52722b38572
<pre>
Web Inspector: Console: internal properties are not greyed out in object previews
<a href="https://bugs.webkit.org/show_bug.cgi?id=255385">https://bugs.webkit.org/show_bug.cgi?id=255385</a>

Reviewed by Patrick Angle.

We should have a consistent style between previews and actual remmote object views.

Otherwise, internal properties appear as regular properties in the preview, but then look different when the object view is expanded.

* Source/JavaScriptCore/inspector/protocol/Runtime.json:
* Source/JavaScriptCore/inspector/InjectedScriptSource.js:
(RemoteObject.prototype._appendPropertyPreview.appendPreview):
* Source/WebInspectorUI/UserInterface/Models/PropertyPreview.js:
(WI.PropertyPreview):
(WI.PropertyPreview.fromPayload):
(WI.PropertyPreview.prototype.get private): Added.
Include an `isPrivate` property in `Runtime.PropertyPreview`.
Drive-by: Refactor `WI.PropertyPreview` to put all optional properties into a `{...} = {}`.

* Source/WebInspectorUI/UserInterface/Views/ObjectPreviewView.js:
(WI.ObjectPreviewView.prototype._appendPropertyPreviews):
* Source/WebInspectorUI/UserInterface/Views/ObjectPreviewView.css:
(.object-preview .name:is(.private, .internal)): Added.
Style private and internal property previews.

* LayoutTests/inspector/runtime/getPreview.html:
* LayoutTests/inspector/runtime/getPreview-expected.txt:

Canonical link: <a href="https://commits.webkit.org/262924@main">https://commits.webkit.org/262924@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f6c7bcc7c247420aba34a3af41c9ae3f86765d5c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/3037 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/3102 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/3208 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/4449 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/3431 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/3181 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/3142 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/2652 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/3067 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/3419 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/2726 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/4244 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/917 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/2705 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/2550 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/2521 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/2691 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/2756 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/3979 "run-api-tests-without-change (failure)") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/2883 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/3105 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/2494 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/3106 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/2729 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/2707 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/746 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/743 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/2721 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/3186 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/2924 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/873 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

patrickangle reviewed (approved):
Looks good to me once we have a set of test passes. Looks like a lot of tests were relying on `PropertyPreview` having undefined values for various things instead of `null`/`false` default values and need rebaselined or updated to not output those nulls and falses 😓

dcrousso committed:
Web Inspector: Console: internal properties are not greyed out in object previews
https://bugs.webkit.org/show_bug.cgi?id=255385

Reviewed by Patrick Angle.

We should have a consistent style between previews and actual remmote object views.

Otherwise, internal properties appear as regular properties in the preview, but then look different when the object view is expanded.

* Source/JavaScriptCore/inspector/protocol/Runtime.json:
* Source/JavaScriptCore/inspector/InjectedScriptSource.js:
(RemoteObject.prototype._appendPropertyPreview.appendPreview):
* Source/WebInspectorUI/UserInterface/Models/PropertyPreview.js:
(WI.PropertyPreview):
(WI.PropertyPreview.fromPayload):
(WI.PropertyPreview.prototype.get private): Added.
Include an `isPrivate` property in `Runtime.PropertyPreview`.
Drive-by: Refactor `WI.PropertyPreview` to put all optional properties into a `{...} = {}`.

* Source/WebInspectorUI/UserInterface/Views/ObjectPreviewView.js:
(WI.ObjectPreviewView.prototype._appendPropertyPreviews):
* Source/WebInspectorUI/UserInterface/Views/ObjectPreviewView.css:
(.object-preview .name:is(.private, .internal)): Added.
Style private and internal property previews.

* LayoutTests/inspector/runtime/getPreview.html:
* LayoutTests/inspector/runtime/getPreview-expected.txt:

Canonical link: https://commits.webkit.org/262924@main
